Shared/Samples/Create mobile geodatabase/CreateMobileGeodatabaseView.Model.swift

```diff
@@ -0,0 +1,215 @@
+// Copyright 2023 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+extension CreateMobileGeodatabaseView {
+    /// The view model for the sample.
+    @MainActor
+    class Model: ObservableObject {
+        // MARK: Properties
+        
+        /// A map with a topographic basemap centered on Harpers Ferry, West Virginia, USA.
+        let map: Map = {
+            let map = Map(basemapStyle: .arcGISTopographic)
+            map.initialViewpoint = Viewpoint(latitude: 39.3238, longitude: -77.7332, scale: 1e4)
+            return map
+        }()
+        
+        /// A URL to a temporary geodatabase.
+        private let geodatabaseURL: URL
+        
+        /// A URL to a temporary directory to store the geodatabase.
+        private let directoryURL: URL
+        
+        /// The mobile geodatabase.
+        private var geodatabase: Geodatabase?
+        
+        /// The feature table in the geodatabase.
+        private var featureTable: GeodatabaseFeatureTable?
+        
+        /// The table description used to create a new feature table in the geodatabase.
+        private let tableDescription: TableDescription = {
+            // Create a description for the feature table.
+            let description = TableDescription(
+                name: "LocationHistory",
+                spatialReference: .wgs84,
+                geometryType: Point.self
+            )
+            
+            // Create and add the field descriptions for the table.
+            description.addFieldDescriptions([
+                // `FieldType.OID` is the primary key of the SQLite table.
+                FieldDescription(name: "oid", fieldType: .oid),
+                // `FieldType.DATE` is the date column used to store a calendar date.
+                FieldDescription(name: "collection_timestamp", fieldType: .date)
+            ])
+            
+            // Set unnecessary properties to false.
+            description.hasAttachments = false
+            description.hasM = false
+            description.hasZ = false
+            
+            return description
+        }()
+        
+        /// The list of features in the feature table.
+        @Published private(set) var features: [FeatureItem] = []
+        
+        /// The count of features in the feature table.
+        @Published private(set) var featureCount = 0
```

Comment:
Can `featureCount` be removed and callers can instead just derive this information from `features`?

`Model.updateFeatures()` will need to run on every new feature creation though.

Reply:
Yes, we could do that. All the other samples use a [string instead of a count](url). We could also just add features to the list as they are created in `Model.addFeature(mapPoint:)`. The downside to that is we don't get to showcase querying the feature table. Then again we have a [dedicated sample](https://github.com/Esri/arcgis-maps-sdk-swift-samples/tree/main/Shared/Samples/Query%20feature%20table) for that. Do you think we should get rid of both the `featureCount` and `Model.queryFeatures()`?

Reply:
If there's a dedicated sample for querying a feature table and it isn't vital to the functionality being sampled, yes, I'd probably opt to just keep the `features` list updated as they're made.